src/datumaro/plugins/sam_transforms/interpreters/sam_decoder_for_amg.py

```diff
@@ -0,0 +1,219 @@
+    """
+
+    points: np.ndarray = field()
+
+
+@attrs(slots=True, kw_only=True, order=False)
+class AMGMasks(Annotation):
+    """Intermediate annotation class for SAM decoder outputs.
+
+    Attributes:
+        masks: Array of masks corresponded to the points.
+        iou_preds: Array of Intersection over Union (IoU)
+            prediction scores corresponded to the points.
+    """
+
+    masks: np.ndarray = field()
+    iou_preds: np.ndarray = field()
+
+    @classmethod
+    def cat(cls, masks: List["AMGMasks"]) -> "AMGMasks":
+        """Concatenate a list of `AMGMasks` into a single `AMGMasks` object.
+
+        Parameters:
+            masks: List of `AMGMasks` to concatenate.
+
+        Returns:
+            A new AMGMasks containing the concatenated masks and IoU prediction scores.
+        """
+        return AMGMasks(
+            masks=np.concatenate([mask.masks for mask in masks], axis=0),
+            iou_preds=np.concatenate([mask.iou_preds for mask in masks], axis=0),
+        )
+
+    def postprocess(
+        self,
+        mask_threshold: float,
+        pred_iou_thresh: float,
+        stability_score_offset: float,
+        stability_score_thresh: float,
+        box_nms_thresh: float,
+        min_mask_region_area: int,
+    ) -> List[Mask]:
+        """Postprocesses the masks with the given parameters.
+
+        Parameters:
+            pred_iou_thresh (float): A filtering threshold in [0,1], using the
+                model's predicted mask quality.
+            stability_score_thresh (float): A filtering threshold in [0,1], using
+                the stability of the mask under changes to the cutoff used to binarize
+                the model's mask predictions.
+            stability_score_offset (float): The amount to shift the cutoff when
+                calculated the stability score.
+            box_nms_thresh (float): The box IoU cutoff used by non-maximal
+                suppression to filter duplicate masks.
+            min_mask_region_area (int): If >0, postprocessing will be applied
+                to remove the binary mask which has the number of 1s less than min_mask_region_area.
+
+        Returns:
+            List of :class:`Mask`s representing the postprocessed masks.
+        """
+        masks, iou_preds = self.masks, self.iou_preds
+
+        if pred_iou_thresh > 0.0:
+            keep_mask = iou_preds > pred_iou_thresh
+            masks = masks[keep_mask]
+            iou_preds = iou_preds[keep_mask]
+
+        if stability_score_thresh > 0.0:
+            keep_mask = (
+                self._calculate_stability_score(
+                    masks=masks,
+                    mask_threshold=mask_threshold,
+                    stability_score_offset=stability_score_offset,
+                )
+                > stability_score_thresh
+            )
+            masks = masks[keep_mask]
+            iou_preds = iou_preds[keep_mask]
+
+        binary_masks = masks > mask_threshold
```

Comment:
Minor comment: `mask_threshold` is fed into `self._calculate_stability_score(...)`, why don't you put `binary_masks = masks > mask_threshold` logic into `self._calculate_stability_score(...)`?

Reply:
The function `self._calculate_stability_score(...)` generates a `keep_mask`, which is a 1D boolean mask indicating what to retain after post-processing `masks` (an array of 2D images with a single floating-point number channel produced by the SAM decoder). A similar pattern of using `keep_mask` for post-processing is maintained in L91-93 and L110-112. So, the code flow seems unnatural if we let `self._calculate_stability_score(...)` to output `binary_masks`.

Reply:
Understood. Thanks.